runtime-modules/storage/src/lib.rs

```diff
@@ -3942,4 +3602,342 @@ impl<T: Trait> Module<T> {
             distribution_bucket_index,
         }
     }
+
+    fn generate_new_storage_buckets(
+        bucket_ids: &BTreeSet<T::StorageBucketId>,
+        new_voucher_update: VoucherUpdate,
+        op: &BagOperationParams<T>,
+    ) -> Result<BTreeMap<T::StorageBucketId, StorageBucket<T>>, Error<T>> {
+        // create temporary iterator of bucket
+        let tmp = bucket_ids
+            .iter()
+            // discard non existing bucket and build a (bucket_id, bucket) map
+            .filter_map(|id| {
+                Self::ensure_storage_bucket_exists(&id)
+                    .map(|bk| (*id, bk))
+                    .ok()
+            })
+            // attempt to update bucket voucher
+            .map(|(id, bk)| {
+                bk.voucher
+                    .clone()
+                    .update(new_voucher_update)
+                    .map_or_else(
+                        |e| match e {
+                            "size" => Err(Error::<T>::StorageBucketObjectSizeLimitReached),
```

Comment:
Please, refactor "magic consts" in favor of the separate function(s).

Reply:
:+1: